product_attribute_company_favorite/models/product_attribute.py

```diff
@@ -0,0 +1,78 @@
+from odoo import api, fields, models
+
+
+class ProductAttributeIsFavoriteMixin(models.AbstractModel):
+    _name = "product.attribute.favorite.mixin"
+    _description = """Methods used both in product.attribute and
+    product.attribute.value to implement is_favorite functionalities
+    """
+
+    def _name_search(
+        self, name, args=None, operator="ilike", limit=100, name_get_uid=None
+    ):
+        args = list(args or [])
+        args += [("is_favorite", "=", True)]
+        return super()._name_search(
+            name, args=args, operator=operator, limit=limit, name_get_uid=name_get_uid
+        )
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        new_record_favorite_for_all_companies = (
+            self.env["ir.config_parameter"]
+            .sudo()
+            .get_param(self.CREATION_ACROSS_COMPANY_CONFIG)
+        )
+        records = super().create(vals_list)
+        if new_record_favorite_for_all_companies:
+            company_ids = (
+                self.env["res.company"].with_context(active_test=False).search([]).ids
+            )
+            for record in records:
+                for company_id in company_ids:
+                    contextual_record = record.with_company(company_id)
+                    contextual_record.is_favorite = True
+        return records
+
+
+class ProductAttribute(models.Model):
+    _inherit = ["product.attribute", "product.attribute.favorite.mixin"]
+    _name = "product.attribute"
+
+    CREATION_ACROSS_COMPANY_CONFIG = (
+        "product_attribute_company_favorite." "new_attribute_favorite_for_all_companies"
+    )
+
+    is_favorite = fields.Boolean(
+        string="Favorite",
+        company_dependent=True,
+        default=True,
+        help="""If checked, the attribute will be shown
+        when searching attributes in a drop-down list""",
+    )
+
+
+class ProductTemplateAttributeLine(models.Model):
+    _inherit = "product.template.attribute.line"
+
+    is_favorite = fields.Boolean(
+        related="attribute_id.is_favorite",
+    )
+
+
+class ProductAttributeValue(models.Model):
+    _inherit = ["product.attribute.value", "product.attribute.favorite.mixin"]
+    _name = "product.attribute.value"
+
+    is_favorite = fields.Boolean(
```

Comment:
I think you can set is_favorite in the mixin.

Reply:
OK, I wasn't sure because the help message was not the same, but I changed it to be more generic. Done